apps/notifications/services/notification.py

```diff
@@ -0,0 +1,48 @@
+from apps.subscribers.models.subscriber import Subscriber
+from apps.notifications.models.payload import Payload
+import os
+import json
+from pushy_notification.celery import celery_app
+from django.core.exceptions import ObjectDoesNotExist
+
+
+class NotificationService:
+    def _create_subscription_info_dict(self, subscriber):
+        subscription_info = {
+            'endpoint': subscriber.endpoint_url,
+            'keys': {
+                'p256dh': subscriber.p256dh_key,
+                'auth': subscriber.auth_key
+            }
+        }
+        return subscription_info
+
+    def _create_notification_payload_json(self, payload):
+        payload_data = {
+            'title': payload.title,
+            'message': payload.body
+        }
+        return json.dumps(payload_data)
+
+    def _create_vapid_claims_dict(self):
+        vapid_claims = {
+            'sub': 'mailto:john@doe.com'
+        }
+        return vapid_claims
+
+    def send_bulk_notifications(self, subscriber_ids, payload_id):
+        try:
+            for subscriber_id in subscriber_ids:
+                subscriber = Subscriber.objects.get(subscriber_id=subscriber_id)
+                payload = Payload.objects.get(payload_id=payload_id)
+                subscription_info = self._create_subscription_info_dict(
+                                        subscriber=subscriber)
```

Comment:
`SubscriberService:
	def get_by_id(id):
		try:
			s = Subs.objs.get(id=id)
		except Subscriber.ObjectDoesNotExist as e:
			return None`

Reply:
Use SubscriberService here. DB should be accessed only from service layer

Reply:
Let's add only the part of code that can raise exception under try block

Reply:
One invalid subscriber id breaks the entire flow. We can try to avoid it